acceptance/tests/connect/connect_inject_test.go

```diff
@@ -37,97 +33,7 @@ func TestConnectInject(t *testing.T) {
-				podList, err := ctx.KubernetesClient(t).CoreV1().Pods(ctx.KubectlOptions(t).Namespace).List(context.Background(), metav1.ListOptions{
-					LabelSelector: labelSelector,
-				})
-				require.NoError(t, err)
-				require.Len(t, podList.Items, 1)
-				require.Len(t, podList.Items[0].Spec.Containers, 2)
-			}
-
-			if c.secure {
-				logger.Log(t, "checking that the connection is not successful because there's no intention")
-				if cfg.EnableTransparentProxy {
-					k8s.CheckStaticServerConnectionFailing(t, ctx.KubectlOptions(t), "http://static-server")
-				} else {
-					k8s.CheckStaticServerConnectionFailing(t, ctx.KubectlOptions(t), "http://localhost:1234")
-				}
-
-				logger.Log(t, "creating intention")
-				_, err := consulClient.Connect().IntentionUpsert(&api.Intention{
-					SourceName:      staticClientName,
-					DestinationName: staticServerName,
-					Action:          api.IntentionActionAllow,
-				}, nil)
-				require.NoError(t, err)
-			}
-
-			logger.Log(t, "checking that connection is successful")
-			if cfg.EnableTransparentProxy {
-				// todo: add an assertion that the traffic is going through the proxy
-				k8s.CheckStaticServerConnectionSuccessful(t, ctx.KubectlOptions(t), "http://static-server")
-			} else {
-				k8s.CheckStaticServerConnectionSuccessful(t, ctx.KubectlOptions(t), "http://localhost:1234")
-			}
-
-			// Test that kubernetes readiness status is synced to Consul.
-			// Create the file so that the readiness probe of the static-server pod fails.
-			logger.Log(t, "testing k8s -> consul health checks sync by making the static-server unhealthy")
-			k8s.RunKubectl(t, ctx.KubectlOptions(t), "exec", "deploy/"+staticServerName, "--", "touch", "/tmp/unhealthy")
-
-			// The readiness probe should take a moment to be reflected in Consul, CheckStaticServerConnection will retry
-			// until Consul marks the service instance unavailable for mesh traffic, causing the connection to fail.
-			// We are expecting a "connection reset by peer" error because in a case of health checks,
-			// there will be no healthy proxy host to connect to. That's why we can't assert that we receive an empty reply
-			// from server, which is the case when a connection is unsuccessful due to intentions in other tests.
-			logger.Log(t, "checking that connection is unsuccessful")
-			if cfg.EnableTransparentProxy {
-				k8s.CheckStaticServerConnectionMultipleFailureMessages(t, ctx.KubectlOptions(t), false, []string{"curl: (56) Recv failure: Connection reset by peer", "curl: (52) Empty reply from server", "curl: (7) Failed to connect to static-server port 80: Connection refused"}, "http://static-server")
-			} else {
-				k8s.CheckStaticServerConnectionMultipleFailureMessages(t, ctx.KubectlOptions(t), false, []string{"curl: (56) Recv failure: Connection reset by peer", "curl: (52) Empty reply from server"}, "http://localhost:1234")
-			}
+			ConnectInject(t, ctx, cfg, c.secure, c.autoEncrypt, false)
```

Comment:
Also thinking out loud here: I wonder if `ConnectInject` is a bit vague of a name and could be renamed to something more descriptive of what it is actually doing (installing a cluster, applications and checking connectivity)?  or maybe something `ConnectInjectBasicInstallTest` or something. *shrug*

Reply:
Oh I like those! Either is fine, thanks!